src/eureka/S3_data_reduction/s3_reduce_niriss.py

```diff
@@ -0,0 +1,399 @@
+        # FINDME: Will want to use DQ array in the future
+        # to flag certain pixels
+        data['mask'] = (['time', 'y', 'x'], np.ones(data.flux.shape,
+                                                    dtype=bool))
+
+        # Check if arrays have NaNs
+        data.mask.values = util.check_nans(data.flux.values,
+                                           data.mask.values,
+                                           log, name='FLUX')
+        data.mask.values = util.check_nans(data.err.values,
+                                           data.mask.values,
+                                           log, name='ERR')
+        data.mask.values = util.check_nans(data.v0.values,
+                                           data.mask.values,
+                                           log, name='V0')
+
+        # Manually mask regions [colstart, colend, rowstart, rowend]
+        if hasattr(meta, 'manmask'):
+            log.writelog("  Masking manually identified bad pixels",
+                         mute=(not meta.verbose))
+            for i in range(len(meta.manmask)):
+                colstart, colend, rowstart, rowend = meta.manmask[i]
+                data['mask'][rowstart:rowend, colstart:colend] = 0
+
+        # Perform outlier rejection of sky background along time axis
+        log.writelog('  Performing background outlier rejection',
+                     mute=(not meta.verbose))
+
+        data = inst.fit_bg(data, meta, readnoise=meta.readnoise,
+                           sigclip=meta.sigclip, box=meta.box,
+                           filter_size=meta.filter_size,
+                           bkg_estimator=meta.bkg_estimator,
+                           testing=meta.testing_S3,
+                           isplots=meta.isplots_S3)
+
+        if meta.isplots_S3 >= 3:
+            log.writelog('  Creating figures for background '
+                         'subtraction', mute=(not meta.verbose))
+            iterfn = range(meta.int_start, meta.n_int)
+            if meta.verbose:
+                iterfn = tqdm(iterfn)
+            for n in iterfn:
+                # make image+background plots
+                plots_s3.image_and_background(data, meta, n, m)
+
+        # Calulate and correct for 2D drift
+        if hasattr(inst, 'correct_drift2D'):
+            log.writelog('  Correcting for 2D drift',
+                         mute=(not meta.verbose))
+            inst.correct_drift2D(data, meta, m)
+
+        # Compute median frame
+        medapdata = np.median(data.flux, axis=0)
+
+        # creates mask for the traces
+        box_masks = niriss_extraction.dirty_mask(medapdata, meta.trace,
+                                                 boxsize1=meta.boxsize1,
+                                                 boxsize2=meta.boxsize2,
+                                                 boxsize3=meta.boxsize3,
+                                                 isplots=meta.isplots_S3)
```

Comment:
These are the lines with the undefined meta parameters that are causing the test and my local runs to fail. Have you been running the class-based method locally, or did you just forget to include boxsize 1-3 in the S3 NIRISS template?

Reply:
I included the boxsizes in the S3 template! Sorry, I forgot to push that (since those files are ignored I can't do it through my typical way). Should be updated now!